lib/pure/random.nim

```diff
@@ -362,7 +362,7 @@ proc rand*[T](r: var Rand; a: openArray[T]): T {.deprecated.} =
   ## Use `sample[T](Rand, openArray[T])<#sample,Rand,openArray[T]>`_ instead.
   result = a[rand(r, a.low..a.high)]
 
-proc rand*[T: SomeInteger](t: typedesc[T]): T =
+proc rand*[T: SomeInteger|range](t: typedesc[T]): T =
```

Comment:
In the failing example this proc was already called with the old signature. Not sure if the addition of `range` is needed? For some other example maybe?

Reply:
You're right, not necessary. Should the docs remain explicit about accepting range parameters?